src/views/Map/AgentMAp.js

```diff
@@ -0,0 +1,173 @@
+    // Try HTML5 geolocation.
+    if (navigator.geolocation) {
+      pos = { lat: 6.60954222424663, lng: 3.370920970651687 };
+      navigator.geolocation.getCurrentPosition((position) => {
+        pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        if (pos !== undefined) {
+          console.log(pos);
+          addRoute();
+        }
+      }, () => {
+        addRoute();
+        handleLocationError(true, infoWindow, map.getCenter());
+      });
+    } else {
+      addRoute();
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
+    // Set the initial Street View camera to the center of the map
+    sv.getPanorama({ location: null, radius: 50 }, processSVData);
+    map.addListener('click', (event) => {
+      sv.getPanorama({ location: event.latLng, radius: 50 }, processSVData);
+    });
+
+  }
+
+  addRoute() {
+    let google = window.google;
+    let start;
+    let directionsRenderer = new google.maps.DirectionsRenderer();
+    let directionsService = new google.maps.DirectionsService();
+    directionsRenderer.setMap(map);
+    directionsRenderer.setPanel(document.getElementById('right-panel'));
+    if (pos === undefined) {
+      start = { lat: 6.60157422424999, lng: 3.360932070651222 };
+    } else {
+      start = pos;
+    }
+    let end = { lat: 6.60157222424999, lng: 3.360920570651222 };
+    directionsService.route({
+      origin: start,
+      destination: end,
+      travelMode: 'DRIVING'
+    }, (response, status) => {
+      if (status === 'OK') {
+        directionsRenderer.setDirections(response);
+      } else {
+        window.alert('Directions request failed due to ' + status);
+      }
+    });
+  }
+
+  processSVData(data, status) {
+    if (status === 'OK') {
+      if (data !== null) {
+        let google = window.google;
+
+        let marker = new google.maps.Marker({
+          position: data.location.latLng,
+          map: map,
+          title: data.location.description
+        });
+      }
+
+      panorama.setPano(data.location.pano);
+      panorama.setPov({
+        heading: 270,
+        pitch: 0
+      });
+      panorama.setVisible(true);
+
+      marker.addListener('click', () => {
+        let markerPanoID = data.location.pano;
+        // Set the Pano to use the passed panoID.
+        panorama.setPano(markerPanoID);
+        panorama.setPov({
+          heading: 270,
+          pitch: 0
+        });
+        panorama.setVisible(true);
+      });
+    } else {
+      console.error('Street View data not found for this location.');
+    }
+  }
+
+  handleLocationError(browserHasGeolocation, infoWindow, pos) {
+    infoWindow.setPosition(pos);
+    infoWindow.setContent(browserHasGeolocation ?
+      'Error: The Geolocation service failed.' :
+      'Error: Your browser doesn\'t support geolocation.');
+    infoWindow.open(map);
+  }
+
+  render() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected 'this' to be used by class method 'render'.](https://app.codacy.com/gh/BuildForSDG/Team-064-Frontend/pullRequest?prid=5703258)